catalog/app/containers/Bucket/PackageDialog/FilesInput.tsx

```diff
@@ -845,15 +591,449 @@ function DropzoneMessage({ error, warn, disabled }: DropzoneMessageProps) {
+  interactive: {
+    cursor: 'pointer',
+  },
+  active: {
+    background: t.palette.action.selected,
+  },
+  err: {
+    borderColor: t.palette.error.main,
+  },
+  warn: {
+    borderColor: t.palette.warning.dark,
+  },
+}))
+
+type ContentsProps = {
+  interactive?: boolean
+  active?: boolean
+  error?: boolean
+  warn?: boolean
+} & React.HTMLAttributes<HTMLDivElement>
+
+function Contents({
+  interactive,
+  active,
+  error,
+  warn,
+  className,
+  ...props
+}: ContentsProps) {
+  const classes = useContentsStyles()
+  return (
+    <div
+      className={cx(
+        className,
+        classes.root,
+        interactive && classes.interactive,
+        active && classes.active,
+        error && classes.err,
+        !error && warn && classes.warn,
+      )}
+      {...props}
+    />
+  )
+}
+
+type FileUploadProps = tagged.ValueOf<typeof FilesEntry.File> & {
+  prefix?: string
+  dispatch: DispatchFilesAction
+}
+
+function FileUpload({ name, state, size, prefix, dispatch }: FileUploadProps) {
+  const path = (prefix || '') + name
+
+  const handle = React.useCallback(
+    (cons: tagged.ConstructorOf<typeof FilesAction>) => (e: React.MouseEvent) => {
+      e.stopPropagation()
+      dispatch(cons(path))
+    },
+    [dispatch, path],
+  )
+
+  const action = React.useMemo(
+    () =>
+      ({
+        added: { hint: 'Remove', icon: 'clear', handler: handle(FilesAction.Revert) },
+        modified: {
+          hint: 'Revert',
+          icon: 'undo',
+          handler: handle(FilesAction.Revert),
+        },
+        hashing: {
+          hint: 'Revert',
+          icon: 'undo',
+          handler: handle(FilesAction.Revert),
+        },
+        deleted: {
+          hint: 'Restore',
+          icon: 'undo',
+          handler: handle(FilesAction.Revert),
+        },
+        unchanged: { hint: 'Delete', icon: 'clear', handler: handle(FilesAction.Delete) },
+      }[state]),
+    [state, handle],
+  )
+
+  const onClick = React.useCallback((e: React.MouseEvent) => {
+    e.stopPropagation()
+  }, [])
+
+  return (
+    // eslint-disable-next-line jsx-a11y/click-events-have-key-events
+    <File
+      onClick={onClick}
+      role="button"
+      tabIndex={0}
+      state={state}
+      name={name}
+      size={size}
+      action={
+        <M.IconButton onClick={action.handler} title={action.hint} size="small">
+          <M.Icon fontSize="inherit">{action.icon}</M.Icon>
+        </M.IconButton>
+      }
+    />
+  )
+}
+
+type DirUploadProps = tagged.ValueOf<typeof FilesEntry.Dir> & {
+  prefix?: string
+  dispatch: DispatchFilesAction
+}
+
+function DirUpload({ name, state, childEntries, prefix, dispatch }: DirUploadProps) {
+  const [expanded, setExpanded] = React.useState(true)
+
+  const toggleExpanded = React.useCallback(
+    (e) => {
+      e.stopPropagation()
+      setExpanded((x) => !x)
+    },
+    [setExpanded],
+  )
+
+  const onClick = React.useCallback((e: React.MouseEvent) => {
+    e.stopPropagation()
```

Comment:
It blocks opening OS's file manager, I suppose? I think you should mention it in comment

Reply:
There is `noClick` option for "react-dropzone". Probably, it will help

Reply:
> It blocks opening OS's file manager, I suppose?

nope, it prevents the click from propagating to parent elements and triggering click handlers there.
adding the comment